tools/wasi/wasi.cpp

```diff
@@ -0,0 +1,213 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "execute.hpp"
+#include "parser.hpp"
+#include <uvwasi.h>
+#include <cassert>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+
+namespace
+{
+// Global state.
+// NOTE: we do not care about uvwasi_destroy(), because it only clears file descriptors (currently)
+// and we are a single-run tool. This may change in the future and should reevaluate.
+uvwasi_t state;
+
+fizzy::ExecutionResult wasi_return_enosys(fizzy::Instance&, fizzy::span<const uint64_t>, int)
+{
+    return UVWASI_ENOSYS;
+}
+
+fizzy::ExecutionResult wasi_proc_exit(fizzy::Instance&, fizzy::span<const uint64_t> args, int)
+{
+    uvwasi_proc_exit(&state, static_cast<uvwasi_exitcode_t>(args[0]));
```

Comment:
This can fail, so maybe trap if not successful?

Reply:
It traps already?

Reply:
It returns `uvwasi_errno_t` whatever it means in this context.